client/web/src/search/input/SearchContextMenu.tsx

```diff
@@ -34,25 +34,77 @@ export interface SearchContextMenuProps extends Omit<SearchContextProps, 'showSe
     closeMenu: () => void
 }
 
+const getFirstMenuItem = (): HTMLButtonElement | null =>
+    document.querySelector('.search-context-menu__item:first-child')
+
 export const SearchContextMenu: React.FunctionComponent<SearchContextMenuProps> = ({
     availableSearchContexts,
     selectedSearchContextSpec,
     defaultSearchContextSpec,
     setSelectedSearchContextSpec,
     closeMenu,
 }) => {
+    const inputElement = useRef<HTMLInputElement | null>(null)
+
     const reset = useCallback(() => {
         setSelectedSearchContextSpec(defaultSearchContextSpec)
         closeMenu()
     }, [closeMenu, defaultSearchContextSpec, setSelectedSearchContextSpec])
 
+    const focusInputElement = (): void => {
+        // Focus the input in the next run-loop to override the browser focusing the first dropdown item
+        // if the user opened the dropdown using a keyboard
+        setTimeout(() => inputElement.current?.focus(), 0)
+    }
+
+    // Triggering the reset button using a keyboard does not trigger a click event (for some reason).
+    // We have to manually check for space and enter keypress and trigger the reset.
+    const onResetButtonKeyDown = useCallback(
+        (event: ReactKeyboardEvent<HTMLButtonElement>): void => {
+            if (event.key === ' ' || event.key === 'Enter') {
+                reset()
+            }
+        },
+        [reset]
+    )
```

Comment:
Reactstrap strikes again. Finally caved in and [filed a bug](https://github.com/reactstrap/reactstrap/issues/2099) on their side. I suggest changing this code to look like this instead:

```suggestion
    // Reactstrap is preventing default behavior on all non-DropdownItem elements inside a Dropdown, 
    // so we need to stop propagation to allow normal behavior (eg. enter and space to activate buttons)
    // See Reactstrap bug: https://github.com/reactstrap/reactstrap/issues/2099
    const onResetButtonKeyDown = useCallback((event: ReactKeyboardEvent<HTMLButtonElement>): void => {
        if (event.key === ' ' || event.key === 'Enter') {
            event.stopPropagation()
        }
    }, [])
```